### PR TITLE
Add new allowConversionToBitmap flag, fixes #773

### LINKED ISSUE #773: Automatically choose between animatedTransformation and transformations if both are set
**Is your feature request related to a problem? Please describe.**
I am trying to load images from user-provided URLs and I don't know ahead of time if the result will be a bitmap or an animated gif.  I want to circle-crop the result either way, but afaict this is not supported.

**Describe the solution you'd like**
If there's an `animatedTransformation` on the request builder and the result is animated, don't apply `transformations`.  If the result is _not_ animated, apply `transitions` normally.

### PR DESCRIPTION
**Pull request by edenman**
This allows animated transformations and normal transformations to
coexist on the same request.  If this flag is set to false:

- A normal bitmap result will have normal transformations applied to it
- An animated result will be left intact which allows
  animatedTransformations to apply.

### CODE REVIEW

**Comment by edenman** [comment]
Wasn't sure how to re-generate `coil-base/api/coil-base.api`...tried assemble_docs.sh but that didn't do it

**Comment by colinrtwhite** [comment]
Awesome, thanks for adding this! You'll need to run `./gradlew apiDump` to updated the API file. Also you'll need to add a new `@Deprecated(HIDDEN)`-annotated constructor and copy function for `Options` to maintain binary compatibility ([like this](https://github.com/coil-kt/coil/blob/main/coil-base/src/main/java/coil/decode/Options.kt#L111)).

File: coil-base/src/main/java/coil/decode/Options.kt [hunk 1]
```diff
@@ -137,6 +144,24 @@ class Options(
...
         diskCachePolicy: CachePolicy = this.diskCachePolicy,
         networkCachePolicy: CachePolicy = this.networkCachePolicy
-    ) = copy(context, config, colorSpace, scale, allowInexactSize, allowRgb565, premultipliedAlpha, headers, parameters,
-        memoryCachePolicy, diskCachePolicy, networkCachePolicy)
+    ) = copy(context, config, colorSpace, scale, allowInexactSize, allowRgb565, premultipliedAlpha,
+        allowConversionToBitmap, headers, parameters, memoryCachePolicy, diskCachePolicy, networkCachePolicy)
+
+
+    @Deprecated(message = "Kept for binary compatibility.", level = DeprecationLevel.HIDDEN)
+    fun copy(
```
Comment (colinrtwhite) [inline coil-base/src/main/java/coil/decode/Options.kt:152]: @edenman We need to add another  deprecated constructor [like this](https://github.com/coil-kt/coil/blob/main/coil-base/src/main/java/coil/decode/Options.kt#L111) as well.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -12,4 +12,6 @@ If you would like to contribute code you can do so through GitHub by forking the
 
 When submitting code, please make every effort to follow existing conventions and style in order to keep the code as readable as possible. Please also make sure your code passes all tests by running `./test.sh`.
 
+If you are making an API change, run `./gradlew apiDump` and include any changed files in your pull request.
```
Comment (colinrtwhite) [inline CONTRIBUTING.md:15]: 🙏

**Comment by colinrtwhite** [comment]
@edenman Looks like there are just a couple minor ktlint warnings left.

**Comment by edenman** [comment]
Also had a test failure on api 19, fixed that and the ktlint stuff

**Review by colinrtwhite — APPROVED** [review]
@edenman Awesome, thanks for implementing this!

**Pull request merged**